internal/utils.go

```diff
@@ -317,6 +320,35 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]interface{
 	return policyTemplates, nil
 }
 
+// This function returns configurationPolicyName with Index+1 based on the provided PolicyConfig.
+// When index is 0, it won't attach any number. When useManifestName is true,
+// it returns the policyConf.Manifests[manifestGroupIndex].Name otherwise,
+// It returns policyConf.Name + index
+func getConfigurationPolicyName(policyConf *types.PolicyConfig, manifestGroupIndex, policyNum,
+	manifestIndex int, useManifestName bool,
+) string {
+	// Do not append a number to configName when it is used for the first time.
+	var configName string
+
+	if useManifestName && policyConf.Manifests[manifestGroupIndex].Name != "" {
+		// If the user does not added Manifests[manifestGroupIndex].Name
+		// then return policyConf.Name as a configurationPolicyName
+		configName = policyConf.Manifests[manifestGroupIndex].Name
+		if manifestIndex > 0 && configName != "" {
+			return fmt.Sprintf("%s%d", configName, manifestIndex+1)
+		}
+	}
+
+	if configName == "" {
+		configName = policyConf.Name
+		if policyNum > 1 {
+			return fmt.Sprintf("%s%d", configName, policyNum)
```

Comment:
(optional) With the helper function, we could move the increment logic here (and it'd match the manifest code above for consistency).
```suggestion
		if policyNum > 0 {
			return fmt.Sprintf("%s%d", configName, policyNum+1)
```